ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
The sentinel was done the way it is because 1) the size of a process name is right now the same size a pointer (if it is larger than a pointer then there would be an issue), and 2) it was a general way to update the code without changing ompi_group_t. I also do not care about 32-bit systems. We haven't had one here ~ 10 years. Loosing a bit in 64 is not that big a deal but I can see why it causes problems with 32-bit.

It will take a rewrite of the ompi_group_t code to do as you suggest but I do not think that will happen in the 2.0.x time frame. It will take some time to write then it needs to soak in master. I do not want to hold up 2.0.0 for this. I am really surprised the issues @ggouaillardet identified did not come up during testing of the dynamic add_procs code (which is now over 5 months old). If we change ompi_group_t we will have to be very careful how we proceed to avoid increasing the memory cost or increasing the cost of ompi_comm_peer_lookup.